src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java

```diff
@@ -235,4 +238,17 @@ public void updateOnTheFlySecurityHotspots(@NotNull Map<VirtualFile, Collection<
       content.setDisplayName(buildTabName(count, SonarLintToolWindowFactory.SECURITY_HOTSPOTS_TAB_TITLE));
     }
   }
+
+  public boolean isSecurityHotspotsTabActive() {
+    var toolWindow = getToolWindow();
+    if (toolWindow != null && securityHotspotsContent != null) {
+      return toolWindow.isVisible() && securityHotspotsContent.isSelected();
+    }
+    return false;
+  }
+
+  @Override
+  public void selectionChanged(@NotNull ContentManagerEvent event) {
+    SonarLintUtils.getService(project, CodeAnalyzerRestarter.class).refreshOpenFiles();
```

Comment:
Maybe a small comment to explain why this is needed ? I think it could be confusing in a while